src/core/connection.c

```diff
@@ -5115,7 +5158,8 @@ QuicConnRecvPostProcessing(
 
     if (Packet->HasNonProbingFrame &&
         Packet->NewLargestPacketNumber &&
-        !(*Path)->IsActive) {
+        !(*Path)->IsActive &&
+        Connection->Paths[0].Route.State == RouteResolved) {
```

Comment:
I think we want to switch regardless. We just received the packet from here, we need to switch.

Reply:
this is some leftover.

Reply:
fixed